src/SetupApi.Desktop/SetupApi+SP_DEVICE_INTERFACE_DETAIL_DATA.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using System.Runtime.InteropServices;
+
+    /// <content>
+    /// Contains the <see cref="SP_DEVICE_INTERFACE_DETAIL_DATA"/> nested type.
+    /// </content>
+    public partial class SetupApi
+    {
+        /// <summary>
+        /// An SP_DEVICE_INTERFACE_DETAIL_DATA structure contains the path for a device interface.
+        /// </summary>
+        [StructLayout(LayoutKind.Sequential)]
```

Comment:
Wow! I've noticed that using `string` in a struct keeps the struct from being 'pinnable'. You're telling me that instead of changing the field to `char*`, I can keep it as `string` and just add this attribute parameter? Awesome!